services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/StorageService.java

```diff
@@ -85,10 +88,12 @@ protected SafeFuture<?> doStart() {
                       serviceConfig.getMetricsSystem(),
                       serviceConfig.getDataDirLayout().getBeaconDataDirectory(),
                       config);
-              database = dbFactory.createDatabase();
-
-              database.migrate();
 
+              try {
+                database = dbFactory.createDatabase();
+              } catch (EphemeryException e) {
+                database = ephemeryDatabaseReset.resetDatabaseAndCreate(serviceConfig, dbFactory);
```

Comment:
silently doing things like this can make it hard to reason about when doing support tickets. 
I'd suggest 
 - logging a message here as a warning - something like `"Ephemery network deposit contract has updated, resetting the stored database and slashing protection data."` or something...
 - moving the variable definition inside this block so we can see its only in this one place that its used.